### ISSUE TITLE
Feature request : hosts file support

### ISSUE THREAD

**Issue by shirojishi** [Jigsaw-Code/intra#31]
Its quite nice have hosts file support for block certain site or ip , and some redirect ip using for website

**Comment by G-9999** [comment]
Same request... I want to block facebook.com and a few others!

**Comment by klemy** [comment]
I agree too, It would be awesome with this king of project which combine multiple source of host block list (reported adware, malware, ads, and few more) : https://github.com/StevenBlack/hosts ! It's work like a charm with my laptop (windows 10 and linux), why not on android?

**Comment by westurner** [comment]
/etc/hosts isn't writeable without rooting.

What would be the fastest (or fast enough) way to do these lookups locally with Android?

**Comment by westurner** [comment]
Would a BloomFilter lookup first be any faster or necessary?
https://google.github.io/guava/releases/23.0-android/api/docs/index.html?com/google/common/hash/BloomFilter.html

Is there something better than ImmutableSortedMap.get()?
https://google.github.io/guava/releases/23.0-android/api/docs/com/google/common/collect/ImmutableSortedMap.html#get-java.lang.Object-

**Comment by FONZACUS** [comment]
hopeful bump
id like to also request this. editing the droids hostfile does get annoying at times, and requires a full reboot whereas pcs can reload it fine after a few mins
to keep the apk file small, itd be awesome to have links to popular host files that download them, and to be able to import our own
